### PR TITLE
Add function to save playlist.

### LINKED ISSUE #263: Feature Request: Save playlist
We would like to have a function to save the playlist when playing.

### PR DESCRIPTION
**Pull request by inflation**
In order to fix #263.

image

### CODE REVIEW

File: iina/MainWindowMenuActions.swift [hunk 1]
```diff
@@ -302,5 +302,23 @@ extension MainWindowController {
     inspector.showWindow(self)
     inspector.updateInfo()
   }
+  
+  @IBAction func menuSavePlaylist(_ sender: NSMenuItem) {
+    let _ = Utility.quickSavePanel(title: "Save to playlist", types: ["m3u8"], ok: {
```
Comment (lhc70000) [inline iina/MainWindowMenuActions.swift]: Please use trailing closure:
```
Utility.quickSavePanel(title: "Save to playlist", types: ["m3u8"]) {
  ...
}

File: iina/MainWindowMenuActions.swift [hunk 2]
```diff
@@ -332,5 +332,23 @@ extension MainWindowController {
...
+        var playlist = ""
+        for item in playerCore.info.playlist {
+          playlist.append((item.filename + "\n"))
+        }
+        
+        do {
+          try playlist.write(to: url, atomically: true, encoding: String.Encoding.utf8)
+        } catch let error as NSError {
+          print("error writing to url \(url)")
+          print(error.localizedDescription)
```
Comment (lhc70000) [inline iina/MainWindowMenuActions.swift]: Please show an alert to user, or at least use `Utility.log`

File: iina/Utility.swift [hunk 3]
```diff
@@ -79,6 +79,21 @@ class Utility {
...
+  static func quickSavePanel(title: String, types: [String], ok: (URL) -> Void) -> Bool {
+    let panel = NSSavePanel()
+    panel.title = title
+    panel.canCreateDirectories = true
+    panel.allowedFileTypes = types
+    if panel.runModal() == NSFileHandlingPanelOKButton {
+      if let url = panel.url {
+        ok(url)
+      }
+      return true
```
Comment (lhc70000) [inline iina/Utility.swift]: Better return `false` when `url` is `nil`?

**Review by lhc70000 — CHANGES_REQUESTED** [review]

**Comment by alejx** [comment]
A suggestion. What about create a new file format for IINA playlist instead of m3u8? Just in case of adding any future fancy functions into the playlist.

**Comment by alejx** [comment]
Add a keyboard shortcut like `cmd+shift+s`?

File: iina/Utility.swift (hunk 3 above)
Comment (inflation) [inline iina/Utility.swift]: For these panels, we don't check the return value anyway. Checking url's non-emptiness seems to be done all in the callbacks.

**Comment by inflation** [comment]
@alejx I would more like to keep with the open standard. You don't want users to find the exported playlist cannot be opened elsewhere.

**Review by lhc70000 — APPROVED** [review]

**Pull request merged**